osm_cli/cli/main.py

```diff
@@ -13,6 +14,30 @@ def osm():
 @click.argument("file_path", type=click.Path(exists=True))
 @click.argument("file_id", type=str)
 def pdf_xml(file_path, file_id):
-    """This function converts a file from PDF to XML"""
-    # Function Implementation
+    """
+    Args:
+        file_path (file path): First parameter
+        file_id (string): Second parameter
+    Returns:
+        Creates an XML file in the directory xmls_sciencebeam
+    """
+    try:
+        converter = PDFConverter()
+
+        if not converter.is_docker_running():
+            raise click.ClickException('Please make sure the docker is running')
+
+        if not converter.is_host_ready():
+            raise click.ClickException('The converter server is offline')
+
+        xml_content = converter.convert(file_path)
```

Comment:
can we keep the cli functions concise. They should just call out to functions that do the heavy lifting. So the functionality should be encapsulated into _pdf_xml or something like that, stored in a different module, and tested independent of the cli.

Reply:
We have  encapsulated the functionality into a pdf -> xml conversion function in a separate module.